cnxepub/formatters.py

```diff
@@ -317,15 +319,36 @@ def _fix_namespaces(html):
     return etree.tostring(let, pretty_print=True, encoding='utf-8')
 
 
-def exercise_callback_factory(match, url_template):
+def _replace_tex_math(node, mml_url):
+    """call mml-api service to replace TeX math in body of node with mathml"""
+
+    res = requests.post(mml_url, {'math': node.text,
+                                  'mathType': 'TeX',
+                                  'mml': 'true'})
+    if res:
+        eq = res.json()
+        mml = etree.fromstring(eq['components'][0]['source'])
+        if node.tag.endswith('span'):
+            mml.set('display', 'inline')
+        elif node.tag.endswith('div'):
+            mml.set('display', 'block')
+        return mml
+    else:
+        return None
+
+
+def exercise_callback_factory(match, url_template, token=None, mml_url=None):
     """Create a callback function to replace an exercise by fetching from
     a server."""
-    import requests
 
     def _replace_exercises(elem):
         item_code = elem.get('href')[len(match):]
         url = url_template.format(itemCode=item_code)
-        res = requests.get(url)
+        if token:
+            headers = {'Authorization': 'Bearer {}'.format(token)}
+            res = requests.get(url, headers=headers)
+        else:
+            res = requests.get(url)
         if res:
```

Comment:
Not part of this pull request, but again this is probably not necessary. I guess it's something I missed in a past PR. 😞